File: data/recomposed_projects/llama-3-70b-instruct/signature/commons-codec/evosuite-test/src/test/org/apache/commons/codec/net/BCodec_ESTest.py
from __future__ import annotations
import time
import re
import os
import typing
from typing import *
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.codec.CodecPolicy import *
from src.main.org.apache.commons.codec.net.BCodec import *

# from src.test.org.apache.commons.codec.net.BCodec_ESTest_scaffolding import *
# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *


class BCodec_ESTest(unittest.TestCase):

    def test55(self) -> None:
        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.getDefaultCharset()
        self.assertEqual("UTF-8", string0)

    def test52(self) -> None:
        bCodec0 = BCodec.BCodec2("UTF-8")
        self.assertEqual("UTF-8", bCodec0.getDefaultCharset())

    def test49(self) -> None:
        bCodec0 = BCodec.BCodec0()
        boolean0 = bCodec0.isStrictDecoding()
        self.assertFalse(boolean0)

    def test48(self) -> None:

        charset0 = "UTF-8"
        codecPolicy0 = CodecPolicy.STRICT
        bCodec0 = BCodec(charset0, codecPolicy0)
        boolean0 = bCodec0.isStrictDecoding()
        self.assertTrue(boolean0)

    def test47(self) -> None:

        bCodec0 = BCodec.BCodec0()
        byteArray0 = bCodec0._doEncoding(None)
        self.assertIsNone(byteArray0)

    def test46(self) -> None:
        bCodec0 = BCodec.BCodec0()
        byteArray0 = bCodec0._doDecoding(None)
        self.assertIsNone(byteArray0)

    def test45(self) -> None:
        bCodec0 = BCodec.BCodec0()
        charset0 = "UTF-8"
        string0 = bCodec0.encode0(None, charset0)
        self.assertIsNone(string0)

    def test44(self) -> None:

        bCodec0 = BCodec.BCodec0()

        try:
            bCodec0.encode1("org.apache.commons.codecCodcPolicy", "")
            self.fail("Expecting exception: IllegalCharsetNameException")

        except UnicodeEncodeError as e:
            self.verifyException("java.nio.charset.Charset", e)

    def test43(self) -> None:
        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.encode1(None, None)
        self.assertIsNone(string0)

    def test42(self) -> None:
        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.encode2(None)
        self.assertIsNone(string0)

    def test41(self) -> None:

        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.decode0(None)
        self.assertIsNone(string0)

    def test40(self) -> None:
        bCodec0 = BCodec.BCodec0()
        object0 = bCodec0.encode3(None)
        self.assertIsNone(object0)

    def test38(self) -> None:

        bCodec0 = BCodec.BCodec0()
        object0 = bCodec0.decode1(None)
        self.assertIsNone(object0)

    def test37(self) -> None:

        bCodec0 = BCodec.BCodec0()
        object0 = bCodec0.decode1("=?UTF-8?B??=")
        self.assertEqual("", object0)

    def test36(self) -> None:
        bCodec0 = BCodec.BCodec0()
        charset0 = bCodec0.getCharset()
        self.assertEqual("UTF-8", charset0)

    def test34(self) -> None:
        bCodec0 = BCodec.BCodec0()
        charset0 = "UTF-8"
        string0 = bCodec0.encode0("", charset0)
        assert string0 is not None
        self.assertEqual("=?UTF-8?B??=", string0)

    def test33(self) -> None:
        bCodec0 = BCodec.BCodec1(None)
        try:
            bCodec0.encode2("|/VzA^TOX{m:Fc^$h2")
            self.fail("Expecting exception: RuntimeError")
        except Exception as e:
            pass

    def test32(self) -> None:

        bCodec0 = BCodec.BCodec0()
        try:
            bCodec0.decode0("")
            self.fail("Expecting exception: Exception")

        except Exception as e:
            #
            # RFC 1522 violation: malformed encoded content
            #
            self.verifyException("org.apache.commons.codec.net.RFC1522Codec", e)

    def test31(self) -> None:

        bCodec0 = BCodec.BCodec0()
        object0 = object()
        try:
            bCodec0.encode3(object0)
            self.fail("Expecting exception: Exception")

        except Exception as e:
            # Objects of type java.lang.Object cannot be encoded using BCodec
            self.verifyException("org.apache.commons.codec.net.BCodec", e)

    def test30(self) -> None:
        bCodec0 = BCodec.BCodec1(None)
        try:
            bCodec0.encode3("QYieMAk<yvxSkf")
            self.fail("Expecting exception: RuntimeError")
        except Exception as e:
            self.assertEqual(
                str(e), "Objects of type NoneType cannot be encoded using BCodec"
            )

    def test29(self) -> None:

        bCodec0 = BCodec.BCodec0()
        try:
            bCodec0.decode1(bCodec0)
            self.fail("Expecting exception: Exception")

        except Exception as e:
            #
            # Objects of type org.apache.commons.codec.net.BCodec cannot be decoded using BCodec
            #
            self.verifyException("org.apache.commons.codec.net.BCodec", e)

    def test28(self) -> None:

        # Undeclared exception
        try:
            BCodec.BCodec2(None)
            self.fail("Expecting exception: ValueError")

        except ValueError as e:
            # Null charset name
            self.verifyException("java.nio.charset.Charset", e)

    def test27(self) -> None:

        # Undeclared exception
        try:
            BCodec.BCodec2("|8")
            self.fail("Expecting exception: IllegalCharsetNameException")

        except IllegalCharsetNameException as e:
            #
            # |8
            #
            self.verifyException("java.nio.charset.Charset", e)

    def test26(self) -> None:

        # Undeclared exception
        try:
            BCodec.BCodec2("mXPI")
            self.fail("Expecting exception: UnsupportedCharsetException")

        except UnsupportedCharsetException as e:
            # mXPI
            self.verifyException("java.nio.charset.Charset", e)

    def test24(self) -> None:

        charset0 = "UTF-8"
        codecPolicy0 = CodecPolicy.STRICT
        bCodec0 = BCodec(charset0, codecPolicy0)
        byteArray0 = bytearray(11)
        byteArray0[1] = 86
        # Undeclared exception in Java code
        try:
            bCodec0._doDecoding(byteArray0)
            self.fail("Expecting exception: ValueError")

        except ValueError as e:
            # Strict decoding: Last encoded character (before the paddings if any) is a valid base 64 alphabet but not a possible encoding. Decoding requires at least two trailing 6-bit characters to create bytes.
            self.assertTrue("org.apache.commons.codec.binary.Base64" in str(e))

    def test21(self) -> None:
        bCodec0 = BCodec.BCodec0()
        try:
            bCodec0.encode0("", None)
            self.fail("Expecting exception: RuntimeError")
        except Exception as e:
            pass

    def test20(self) -> None:
        bCodec0 = BCodec.BCodec0()
        try:
            bCodec0.encode1("", None)
            self.fail("Expecting exception: ValueError")
        except ValueError as e:
            self.verifyException("java.nio.charset.Charset", e)

    def test19(self) -> None:

        bCodec0 = BCodec.BCodec0()

        with pytest.raises(UnsupportedCharsetException):
            bCodec0.encode1("A", "A")

        verifyException("java.nio.charset.Charset", UnsupportedCharsetException)

    def test18(self) -> None:

        bCodec0 = BCodec.BCodec1(None)
        try:
            bCodec0.getDefaultCharset()
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            self.assertEqual(str(e), "")
            self.assertEqual(type(e), RuntimeError)

    def test14(self) -> None:

        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.decode0("=?UTF-8?B??=")
        self.assertEqual("", string0)

    def test13(self) -> None:

        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.decode0("=?UTF-8?B?Xyt9XU4iQm03Ki0=?=")
        self.assertEqual('_+}]N"Bm7*-', string0)

    def test12(self) -> None:
        bCodec0 = BCodec.BCodec0()
        byteArray0 = bytearray(5)
        byteArray1 = bCodec0._doDecoding(byteArray0)
        self.assertEqual(0, len(byteArray1))

    def test11(self) -> None:

        charset0 = "UTF-8"
        codecPolicy0 = CodecPolicy.LENIENT
        bCodec0 = BCodec(charset0, codecPolicy0)
        byteArray0 = bytearray(8)
        byteArray0[1] = 73
        byteArray0[6] = 118
        byteArray1 = bCodec0._doDecoding(byteArray0)
        self.assertEqual(1, len(byteArray1))

    def test10(self) -> None:

        bCodec0 = BCodec.BCodec0()
        byteArray0 = bytearray()
        byteArray1 = bCodec0.doEncoding(byteArray0)
        self.assertEqual(bytearray(), byteArray1)

    def test09(self) -> None:

        bCodec0 = BCodec.BCodec0()
        byteArray0 = bytearray(2)
        byteArray1 = bCodec0.doEncoding(byteArray0)
        self.assertIsNot(byteArray1, byteArray0)

    def test04(self) -> None:
        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.encode1("UTF-8", "UTF-8")
        assert string0 is not None
        self.assertEqual("=?UTF-8?B?VVRGLTg=?=", string0)

    def test03(self) -> None:
        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0.encode2('_+}]N"Bm7*-')
        self.assertEqual("=?UTF-8?B?Xyt9XU4iQm03Ki0=?=", string0)
        self.assertIsNotNone(string0)

    def test02(self) -> None:
        bCodec0 = BCodec.BCodec0()
        object0 = bCodec0.encode3("")
        self.assertEqual("=?UTF-8?B??=", object0)
        self.assertIsNotNone(object0)

    def test01(self) -> None:

        codecPolicy0 = CodecPolicy.LENIENT
        bCodec0 = BCodec(None, codecPolicy0)
        charset0 = bCodec0.getCharset()
        self.assertIsNone(charset0)

    def test00(self) -> None:
        bCodec0 = BCodec.BCodec0()
        string0 = bCodec0._getEncoding()
        self.assertEqual("B", string0)

File: java_projects/cleaned_final_projects_evosuite/commons-codec/src/test/java/org/apache/commons/codec/net/BCodec_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.codec.net;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.nio.charset.Charset;
import java.nio.charset.IllegalCharsetNameException;
import java.nio.charset.UnsupportedCharsetException;
import org.apache.commons.codec.CodecPolicy;
import org.apache.commons.codec.net.BCodec;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class BCodec_ESTest extends BCodec_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.getEncoding();
      assertEquals("B", string0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      CodecPolicy codecPolicy0 = CodecPolicy.LENIENT;
      BCodec bCodec0 = new BCodec((Charset) null, codecPolicy0);
      Charset charset0 = bCodec0.getCharset();
      assertNull(charset0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.encode3("");
      assertEquals("=?UTF-8?B??=", object0);
      assertNotNull(object0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.encode2("_+}]N\"Bm7*-");
      assertEquals("=?UTF-8?B?Xyt9XU4iQm03Ki0=?=", string0);
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.encode1("UTF-8", "UTF-8");
      assertNotNull(string0);
      assertEquals("=?UTF-8?B?VVRGLTg=?=", string0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Charset charset0 = Charset.defaultCharset();
      String string0 = bCodec0.encode((String) null, charset0);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Charset charset0 = Charset.defaultCharset();
      String string0 = bCodec0.encode("", charset0);
      assertEquals("=?UTF-8?B??=", string0);
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.encode((String) null);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.encode((Object) null);
      assertNull(object0);
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      byte[] byteArray0 = new byte[2];
      byte[] byteArray1 = bCodec0.doEncoding(byteArray0);
      assertNotSame(byteArray1, byteArray0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      byte[] byteArray0 = new byte[0];
      byte[] byteArray1 = bCodec0.doEncoding(byteArray0);
      assertArrayEquals(new byte[] {}, byteArray1);
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      Charset charset0 = Charset.defaultCharset();
      CodecPolicy codecPolicy0 = CodecPolicy.LENIENT;
      BCodec bCodec0 = new BCodec(charset0, codecPolicy0);
      byte[] byteArray0 = new byte[8];
      byteArray0[1] = (byte)73;
      byteArray0[6] = (byte)118;
      byte[] byteArray1 = bCodec0.doDecoding(byteArray0);
      assertEquals(1, byteArray1.length);
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      byte[] byteArray0 = new byte[5];
      byte[] byteArray1 = bCodec0.doDecoding(byteArray0);
      assertEquals(0, byteArray1.length);
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.decode0("=?UTF-8?B?Xyt9XU4iQm03Ki0=?=");
      assertEquals("_+}]N\"Bm7*-", string0);
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.decode0("=?UTF-8?B??=");
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.decode((String) null);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.decode((Object) null);
      assertNull(object0);
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.decode((Object) "=?UTF-8?B??=");
      assertEquals("", object0);
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec1((Charset) null);
      // Undeclared exception!
      try { 
        bCodec0.getDefaultCharset();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.codec.net.BCodec", e);
      }
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      // Undeclared exception!
      try { 
        bCodec0.encode1("A", "A");
        fail("Expecting exception: UnsupportedCharsetException");
      
      } catch(UnsupportedCharsetException e) {
         //
         // A
         //
         verifyException("java.nio.charset.Charset", e);
      }
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      // Undeclared exception!
      try { 
        bCodec0.encode1("", (String) null);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Null charset name
         //
         verifyException("java.nio.charset.Charset", e);
      }
  }

  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      // Undeclared exception!
      try { 
        bCodec0.encode0("", (Charset) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
      }
  }

  @Test(timeout = 4000)
  public void test22()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec1((Charset) null);
      // Undeclared exception!
      try { 
        bCodec0.encode("");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
      }
  }

  @Test(timeout = 4000)
  public void test23()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec1((Charset) null);
      // Undeclared exception!
      try { 
        bCodec0.encode((Object) "`!9MNQn,#=");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
      }
  }

  @Test(timeout = 4000)
  public void test24()  throws Throwable  {
      Charset charset0 = Charset.defaultCharset();
      CodecPolicy codecPolicy0 = CodecPolicy.STRICT;
      BCodec bCodec0 = new BCodec(charset0, codecPolicy0);
      byte[] byteArray0 = new byte[11];
      byteArray0[1] = (byte)86;
      // Undeclared exception!
      try { 
        bCodec0.doDecoding(byteArray0);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Strict decoding: Last encoded character (before the paddings if any) is a valid base 64 alphabet but not a possible encoding. Decoding requires at least two trailing 6-bit characters to create bytes.
         //
         verifyException("org.apache.commons.codec.binary.Base64", e);
      }
  }

  @Test(timeout = 4000)
  public void test25()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      try { 
        bCodec0.decode("");
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // RFC 1522 violation: malformed encoded content
         //
         verifyException("org.apache.commons.codec.net.RFC1522Codec", e);
      }
  }

  @Test(timeout = 4000)
  public void test26()  throws Throwable  {
      // Undeclared exception!
      try { 
        BCodec.BCodec2("mXPI");
        fail("Expecting exception: UnsupportedCharsetException");
      
      } catch(UnsupportedCharsetException e) {
         //
         // mXPI
         //
         verifyException("java.nio.charset.Charset", e);
      }
  }

  @Test(timeout = 4000)
  public void test27()  throws Throwable  {
      // Undeclared exception!
      try { 
        BCodec.BCodec2("|8");
        fail("Expecting exception: IllegalCharsetNameException");
      
      } catch(IllegalCharsetNameException e) {
         //
         // |8
         //
         verifyException("java.nio.charset.Charset", e);
      }
  }

  @Test(timeout = 4000)
  public void test28()  throws Throwable  {
      // Undeclared exception!
      try { 
        BCodec.BCodec2((String) null);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Null charset name
         //
         verifyException("java.nio.charset.Charset", e);
      }
  }

  @Test(timeout = 4000)
  public void test29()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      try { 
        bCodec0.decode1(bCodec0);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Objects of type org.apache.commons.codec.net.BCodec cannot be decoded using BCodec
         //
         verifyException("org.apache.commons.codec.net.BCodec", e);
      }
  }

  @Test(timeout = 4000)
  public void test30()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec1((Charset) null);
      // Undeclared exception!
      try { 
        bCodec0.encode3("QYieMAk<yvxSkf");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
      }
  }

  @Test(timeout = 4000)
  public void test31()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = new Object();
      try { 
        bCodec0.encode3(object0);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Objects of type java.lang.Object cannot be encoded using BCodec
         //
         verifyException("org.apache.commons.codec.net.BCodec", e);
      }
  }

  @Test(timeout = 4000)
  public void test32()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      try { 
        bCodec0.decode0("");
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // RFC 1522 violation: malformed encoded content
         //
         verifyException("org.apache.commons.codec.net.RFC1522Codec", e);
      }
  }

  @Test(timeout = 4000)
  public void test33()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec1((Charset) null);
      // Undeclared exception!
      try { 
        bCodec0.encode2("|/VzA^TOX{m:Fc^$h2");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
      }
  }

  @Test(timeout = 4000)
  public void test34()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Charset charset0 = Charset.defaultCharset();
      String string0 = bCodec0.encode0("", charset0);
      assertNotNull(string0);
      assertEquals("=?UTF-8?B??=", string0);
  }

  @Test(timeout = 4000)
  public void test35()  throws Throwable  {
      Charset charset0 = Charset.defaultCharset();
      BCodec bCodec0 = BCodec.BCodec1(charset0);
      String string0 = bCodec0.decode("=?UTF-8?B?WnghWyF8aFddKD1ZRTNvSzs=?=");
      assertEquals("Zx![!|hW](=YE3oK;", string0);
  }

  @Test(timeout = 4000)
  public void test36()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Charset charset0 = bCodec0.getCharset();
      assertEquals("UTF-8", charset0.toString());
  }

  @Test(timeout = 4000)
  public void test37()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.decode1("=?UTF-8?B??=");
      assertEquals("", object0);
  }

  @Test(timeout = 4000)
  public void test38()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.decode1((Object) null);
      assertNull(object0);
  }

  @Test(timeout = 4000)
  public void test39()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      try { 
        bCodec0.encode((Object) bCodec0);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Objects of type org.apache.commons.codec.net.BCodec cannot be encoded using BCodec
         //
         verifyException("org.apache.commons.codec.net.BCodec", e);
      }
  }

  @Test(timeout = 4000)
  public void test40()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.encode3((Object) null);
      assertNull(object0);
  }

  @Test(timeout = 4000)
  public void test41()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.decode0((String) null);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test42()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.encode2((String) null);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test43()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.encode1((String) null, (String) null);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test44()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      // Undeclared exception!
      try { 
        bCodec0.encode1("org.apache.commons.codecCodcPolicy", "");
        fail("Expecting exception: IllegalCharsetNameException");
      
      } catch(IllegalCharsetNameException e) {
         //
         // 
         //
         verifyException("java.nio.charset.Charset", e);
      }
  }

  @Test(timeout = 4000)
  public void test45()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Charset charset0 = Charset.defaultCharset();
      String string0 = bCodec0.encode0((String) null, charset0);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test46()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      byte[] byteArray0 = bCodec0.doDecoding((byte[]) null);
      assertNull(byteArray0);
  }

  @Test(timeout = 4000)
  public void test47()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      byte[] byteArray0 = bCodec0.doEncoding((byte[]) null);
      assertNull(byteArray0);
  }

  @Test(timeout = 4000)
  public void test48()  throws Throwable  {
      Charset charset0 = Charset.defaultCharset();
      CodecPolicy codecPolicy0 = CodecPolicy.STRICT;
      BCodec bCodec0 = new BCodec(charset0, codecPolicy0);
      boolean boolean0 = bCodec0.isStrictDecoding();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test49()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      boolean boolean0 = bCodec0.isStrictDecoding();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test50()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.decode("=?UTF-8?B??=");
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test51()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      // Undeclared exception!
      try { 
        bCodec0.encode("", (Charset) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
      }
  }

  @Test(timeout = 4000)
  public void test52()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec2("UTF-8");
      assertEquals("UTF-8", bCodec0.getDefaultCharset());
  }

  @Test(timeout = 4000)
  public void test53()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.encode("");
      assertEquals("=?UTF-8?B??=", string0);
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test54()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      Object object0 = bCodec0.encode((Object) "");
      assertNotNull(object0);
      assertEquals("=?UTF-8?B??=", object0);
  }

  @Test(timeout = 4000)
  public void test55()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      String string0 = bCodec0.getDefaultCharset();
      assertEquals("UTF-8", string0);
  }

  @Test(timeout = 4000)
  public void test56()  throws Throwable  {
      BCodec bCodec0 = BCodec.BCodec0();
      try { 
        bCodec0.decode((Object) bCodec0);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Objects of type org.apache.commons.codec.net.BCodec cannot be decoded using BCodec
         //
         verifyException("org.apache.commons.codec.net.BCodec", e);
      }
  }
}
